src/backend/marsha/core/simple_jwt/tokens.py

```diff
@@ -0,0 +1,265 @@
+"""Specific Marsha JWT models"""
+from django.conf import settings
+from django.utils.translation import gettext_lazy as _
+
+from rest_framework_simplejwt.exceptions import TokenError
+from rest_framework_simplejwt.tokens import AccessToken
+
+from marsha.core.models import NONE, STUDENT
+from marsha.core.simple_jwt.utils import define_locales
+from marsha.core.utils.react_locales_utils import react_locale
+
+
+class LTISelectFormAccessToken(AccessToken):
+    """
+    LTI select form access JWT.
+
+    For now, we stay with the same attributes as the original AccessToken for
+    backward compatibility:
+    ```
+        token_type = "access"
+        lifetime = api_settings.ACCESS_TOKEN_LIFETIME
+    ```
+
+    Note: not usable through our authentication backend since it doesn't provide the
+    `api_settings.USER_ID_CLAIM` information in payload.
+    """
+
+    PAYLOAD_FORM_DATA = "lti_select_form_data"
+
+    @classmethod
+    def for_lti_select_form_data(cls, lti_select_form_data):
+        """
+        Build an LTI Select Form JWT.
+
+        Parameters
+        ----------
+        lti_select_form_data: dict
+            the data to enclose in the JWT's payload.
+
+        Returns
+        -------
+        LTISelectFormAccessToken
+            JWT containing:
+            - lti_select_form_data
+        """
+        token = cls()
+        token.payload[cls.PAYLOAD_FORM_DATA] = lti_select_form_data
+        return token
+
+    def verify(self):
+        """Performs additional validation steps to test payload content."""
+        super().verify()
+
+        if self.PAYLOAD_FORM_DATA not in self.payload:
+            raise TokenError(_("Malformed LTI form token"))
+
+
+class ResourceAccessToken(AccessToken):
+    """
+    Resource dedicated access JWT.
+
+    For now, we stay with the same attributes as the original AccessToken for
+    backward compatibility:
+    ```
+        token_type = "access"
+        lifetime = api_settings.ACCESS_TOKEN_LIFETIME
+    ```
+
+    Note: `api_settings.USER_ID_CLAIM` is currently `resource_id`.
+    """
+
+    @classmethod
+    def for_resource(  # pylint: disable=too-many-arguments
```

Comment:
Reading the method name, I expect to pass as an argument the resource itself.

Reply:
You are right, I will suggest modifications to use the resource instead of the resource ID.